tests/test-integrity.js

```diff
@@ -0,0 +1,110 @@
+var assert = require("assert");
+var fs = require('fs');
+
+var Data = require("./data");
+var moment = require("moment");
+var utils = require("./utils");
+
+
+describe("All data", function() {
+  describe("in sources.js", function() {
+    describe("that have 'release_date' value", function() {
+      it("should have an 'announcement_date' that comes before it", function() {
+        var errors = [];
+        var source;
+
+        for (var i = 0; i < Data.sources.length; i++) {
+          source = Data.sources[i];
+
+          if (source.release_date && source.announcement_date) {
+            if (!moment(source.announcement_date).isBefore(moment(source.release_date))) {
+              errors.push(
+                utils.buildDataHeader(Data, 'sources', i) +
+                ": 'release_date' comes before 'announcement_date'."
```

Comment:
In keeping with the sentence structure of the other tests below, I'd reword this to `": 'announcement_date' should come before 'release_date'."`